h2o-core/src/main/java/water/rapids/ast/prims/filters/dropduplicates/AstDropDuplicates.java

```diff
@@ -0,0 +1,92 @@
+ */
+public class AstDropDuplicates extends AstPrimitive<AstDropDuplicates> {
+  @Override
+  public int nargs() {
+    return 1 + 3;
+  }
+
+  @Override
+  public String[] args() {
+    return new String[]{"ary", "frame", "cols", "droporder"};
+  }
+
+  @Override
+  public Val apply(Env env, Env.StackHelp stk, AstRoot[] asts) {
+    final Frame deduplicatedFrame = stk.track(asts[1].exec(env)).getFrame();
+    final int[] comparedColumnsIndices = parseAndCheckComparedColumnIndices(deduplicatedFrame, stk.track(asts[2].exec(env)));
+    final String dropOrderString = asts[3].str();
+    final KeepOrder keepOrder = EnumUtils.valueOfIgnoreCase(KeepOrder.class, dropOrderString)
+            .orElseThrow(() -> new IllegalArgumentException(String.format("Unknown drop order: '%s'. Known types: %s",
+                    dropOrderString, Arrays.toString(KeepOrder.values()))));
+
+    final DropDuplicateRows dropDuplicateRows = new DropDuplicateRows(deduplicatedFrame, comparedColumnsIndices, keepOrder);
+    final Frame outputFrame = dropDuplicateRows.dropDuplicates();
+    return new ValFrame(outputFrame);
+  }
+
+  private int[] parseAndCheckComparedColumnIndices(final Frame deduplicatedFrame, final Val comparedColumns) {
+    final int[] columnRange;
+
+    if (comparedColumns.isStr()) {
+      final String columnName = comparedColumns.getStr();
+      final int columnIndex = deduplicatedFrame.find(columnName);
+      if (columnIndex == -1) {
+        throw new IllegalArgumentException(String.format("Unknown column name: '%s'", columnName));
+      }
+      columnRange = new int[]{columnIndex};
+    } else if (comparedColumns.isStrs()) {
+      final String[] columnNames = comparedColumns.getStrs();
+      columnRange = new int[columnNames.length];
+
+      for (int i = 0; i < columnNames.length; i++) {
+        final String columnName = columnNames[i];
+        final int columnIndex = deduplicatedFrame.find(columnName);
+        if (columnIndex == -1) {
+          throw new IllegalArgumentException(String.format("Unknown column name: '%s'", columnName));
+        } else if (deduplicatedFrame.types()[columnIndex] == Vec.T_STR || deduplicatedFrame.types()[columnIndex] == Vec.T_BAD
+                || deduplicatedFrame.types()[columnIndex] == Vec.T_UUID) {
+          throw new IllegalArgumentException(String.format("Column '%s' is of unsupported type %s for row de-duplication.", columnName, Vec.TYPE_STR[deduplicatedFrame.types()[columnIndex]]));
+        }
+        columnRange[i] = columnIndex;
+
+      }
+
+    } else if (comparedColumns.isNums()) {
+      final double[] columnRangeDouble = comparedColumns.getNums();
+      columnRange = new int[columnRangeDouble.length];
+      for (int i = 0; i < columnRangeDouble.length; i++) {
+        columnRange[i] = (int) columnRangeDouble[i];
+        if (columnRange[i] < 0 || columnRange[i] > deduplicatedFrame.numCols() - 1) {
```

Comment:
From this check, there could be also list of indices not only a range of indices. Am I correct, if I use an array [1,2,3] but also [2,1,3]? So the second one is not range and still can be used, or not? It is not critical, just quibble. :)

Reply:
This part of code only checks if the given column indices are within deduplicated frame's range. It's a basic sanity check not to try to get a vector with index that is not there. If the index is incorrect, a user-friendly errror is thrown with a sort-of descriptive message.

Reply:
Yes, I understand this part of the code. I just comment here, because it is the first occurrence. But I thought it global. If a user can use also [2,1, 3] order, it is not range, but indices. :) But it is really just note about naming, not about functionality...

Reply:
Oh, I see. Let me fix that m'aam.

Reply:
Fixed. Thank you.